File: examples/java-starter-test/src/main/java/dev/choo/webscraper/Jackson.java
/*
 * Copyright (C) 2025 Steven Choo <info@stevenchoo.nl>
 *
 * MIT License. See the LICENSE file in the root directory or <http://www.opensource.org/licenses/mit-license.php>
 */

package dev.choo.webscraper;

import com.fasterxml.jackson.databind.ObjectMapper;
import lombok.SneakyThrows;

public class Jackson {

    private final ObjectMapper objectMapper = new ObjectMapper();

    @SneakyThrows
    public String serialize() {
        return objectMapper.writeValueAsString(
                User.builder().id("id").name("name").email("email").build());
    }
}

File: examples/kotlin-starter-test/src/main/kotlin/dev/choo/webscraper/Jackson.kt
/*
 * Copyright (C) 2025 Steven Choo <info@stevenchoo.nl>
 *
 * MIT License. See the LICENSE file in the root directory or <http://www.opensource.org/licenses/mit-license.php>
 */

package dev.choo.webscraper

import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.module.kotlin.registerKotlinModule

class Jackson {
    private val objectMapper = ObjectMapper().registerKotlinModule()

    fun serialize(): String =
        objectMapper.writeValueAsString(
            User(id = "id", name = "name", email = "email"),
        )
}
